### PR TITLE
Apply hover class in outline mode

### LINKED ISSUE #27543: is-hovered class no longer applied to blocks
The `is-hovered` css class is no longer being applied to blocks on hover. This means the outlines introduced in outline-mode (https://github.com/WordPress/gutenberg/pull/27499) are no longer visible.

### PR DESCRIPTION
**Pull request by jameskoster**
Closes #27543 

Restores the outline effect on hover when Site Editing:

hover

**To test:**

* Edit a post and ensure the `is-hovered` class is **not** applied when hovering a block
* Zoom out to template view  and ensure the `is-hovered` class **is** applied when hovering a block
* Ensure Navigation mode behaviour is unchanged

### CODE REVIEW

**Comment by david-szabo97** [comment]
A little bit of off-topic: What's the difference between outline mode and navigate mode? 🤔 I'm trying to understand why outline mode is a thing. We already had navigate mode. 

Onto testing the changes.

File: packages/block-editor/src/components/block-list/use-block-props/use-is-hovered.js [hunk 1]
```diff
@@ -7,24 +7,30 @@ import { useSelect } from '@wordpress/data';
...
-			return;
-		}
+	const { isOutlineMode } = useSelect(
+		( select ) => {
+			const { getSettings } = select( 'core/block-editor' );
+			return {
+				isOutlineMode: getSettings().outlineMode,
+			};
+		},
+		[ clientId ]
```
Comment (david-szabo97) [inline packages/block-editor/src/components/block-list/use-block-props/use-is-hovered.js:30]: Is `clientId` necessary as a dependency? It works fine without it.

**Comment by jasmussen** [comment]
> A little bit of off-topic: What's the difference between outline mode and navigate mode? 🤔 I'm trying to understand why outline mode is a thing. We already had navigate mode.

I don't see it as three modes, I see it as two modes, _select_ and _edit_. 

- In select mode, you can press tab 5 times to get to block number 5, which is important for screen readers. It also clicks parent-first through grouped blocks.
- In edit mode, you can click anything to edit it directly.

This remains the same in site and post editors, the only difference is that we are experimenting with also showing the block boundary in _edit mode_ for the site editor. If this is successful it could become an option to enable in either editor.

**Comment by jameskoster** [comment]
Thanks for the review @david-szabo97, I pushed an update.

File: packages/block-editor/src/components/block-list/use-block-props/use-is-hovered.js [hunk 2]
```diff
@@ -20,11 +20,14 @@ export function useIsHovered( ref ) {
 		[]
 	);
 
-	useEffect( () => {
-		if ( ! isNavigationMode ) {
-			return;
-		}
+	const { isOutlineMode } = useSelect( ( select ) => {
```
Comment (david-szabo97) [inline packages/block-editor/src/components/block-list/use-block-props/use-is-hovered.js:23]: Can we use a single `useSelect` for `isNavigationMode` and `isOutlineMode`?

**Review by david-szabo97 — APPROVED** [review]
LGTM! 🚢  Left a minor suggestion.

**Pull request merged**